manipulation/src/main/java/org/jboss/gm/manipulation/ManipulationPlugin.java

```diff
@@ -23,9 +32,45 @@ public void apply(Project project) {
         // add actions to manipulate project
         project.afterEvaluate(new ProjectChangeVersionAction(correspondingModule));
         project.afterEvaluate(new OverrideDependenciesAction(correspondingModule));
-        project.afterEvaluate(new PublishingRepositoryAction());
+        //        project.afterEvaluate(new PublishingRepositoryAction());
         project.afterEvaluate(new ManifestUpdateAction(correspondingModule));
-        project.afterEvaluate(new UploadTaskTransformerAction(correspondingModule));
-        project.afterEvaluate(new PublicationPomTransformerAction(correspondingModule));
+        //        project.afterEvaluate(new UploadTaskTransformerAction(correspondingModule));
+        //        project.afterEvaluate(new PublishingPomTransformerAction(correspondingModule));
+
+        // TODO: add functional tests
+        project.afterEvaluate(new Action<Project>() {
+            @Override
+            public void execute(Project project) {
+                Object preferenceProperty = project.getProperties().get("gm.publishing-plugin");
```

Comment:
Where does this come from please? What happens if its not set?

Reply:
So, I was trying to use a "project property", because I thought I had problem that "system properties" were not correctly set when the gradle daemon was already running, but I can't reproduce it anymore. So I'm going to switch back to system property or will use our ConfigurationFactory.

As for the meaning of the property, the intention was that we would reuse the publishing plugin used by the project (which should be easier for us, we would basically only need to add our publication repo, and the remaining configuration - artifacts to publish, poms, etc - should be already configured by the project). This property is meant to override this behavior.

Btw project properties can be set:
* in project's gradle.properties file, without prefix, ie `gm.publishing-plugin=xxx`,
* via -P switch in the `gradle` command.

Contrary to that "system properties" (obtained via `System.getProperty()`) can be set:
* in gradle.properties files, with "systemProp." prefix, ie `systemProp.gm.publishing-plugin=xxx`
* via -D switch.

Reply:
Ok. I've added a new Configuration class instead of a ConfigFactory.